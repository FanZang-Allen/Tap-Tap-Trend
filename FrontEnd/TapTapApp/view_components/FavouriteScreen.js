import React from 'react';
import {useState, useEffect, useCallback} from 'react';
import {SafeAreaView, StyleSheet} from 'react-native';
import GameFlatList from './GameFlatList';
import GameList from '../model/GameList';
import {baseUrl, user_data} from '../model/Config';

const FavouriteScreen = ({navigation}) => {
  const [refreshing, setRefreshing] = useState(false);
  const [gameData, setgameData] = useState(null);
  const fetchUrl = baseUrl + '/api/game';

  const onRefresh = useCallback(() => {
    setRefreshing(true);
    async function fetchFavourite() {
      const gameClass = new GameList(fetchUrl, user_data.favourite_list);
      await gameClass.FetchData();
      if (gameClass.success === false) {
        setgameData(undefined);
        setRefreshing(false);
      } else {
        setgameData(gameClass.GetData());
        setRefreshing(false);
      }
    }
    fetchFavourite();
  }, [fetchUrl]);

  useEffect(() => {
    let fetchCalled = false;
    async function intialFetch() {
      const gameClass = new GameList(fetchUrl, user_data.favourite_list);
      await gameClass.FetchData();
      if (gameClass.success === false) {
        setgameData(undefined);
      } else {
        setgameData(gameClass.GetData());
      }
    }
    if (fetchCalled === false) {
      intialFetch();
      fetchCalled = true;
    }
  }, [fetchUrl]);

  return (
    <SafeAreaView style={styles.container}>
      <GameFlatList
        gotData={gameData}
        navigation={navigation}
        refreshing={refreshing}
        onRefresh={onRefresh}
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});

export default FavouriteScreen;
