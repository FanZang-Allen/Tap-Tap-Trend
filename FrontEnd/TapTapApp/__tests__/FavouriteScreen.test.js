import 'react-native';
import React from 'react';
import FavouriteScreen from '../view_components/FavouriteScreen';
import renderer from 'react-test-renderer';

test('Profile Rendered Page', () => {
  const snap = renderer.create(<FavouriteScreen />).toJSON();
  expect(snap).toMatchSnapshot();
});
