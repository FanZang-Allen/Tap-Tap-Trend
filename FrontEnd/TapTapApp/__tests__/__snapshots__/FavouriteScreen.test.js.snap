// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Rendered Page 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#9B9B9B",
          "flex": 4,
          "fontSize": 14,
        }
      }
    >
      Loading
    </Text>
  </View>
</RCTSafeAreaView>
`;
